src/index.ts

```diff
@@ -82,19 +82,19 @@ const remarkEmbedder: Plugin<[RemarkEmbedderOptions]> = ({
       }
 
       const {children} = paragraphNode
-      const node = children[0] as Literal & Parent
+      const node = children[0] as Link | Text
```

Comment:
Why the `|`? It’s just a `Link`, presumably, right?

Reply:
L86 is checking if it's a `Text`
```ts
const isText = node.type === 'text'
```

We return early if it's not `Text` **or** if it's an invalid link.
But that seems weird as a node can't be both text & valid link 🤔

@kentcdodds can probably give some more context about this check

Reply:
Yeah, we only *want* to find `link`s but we could find just about anything as the first and only child to a paragraph. But now that I think about it, if `isValidLink` is true, then `isText` will always be false so we could probably remove the `isText` completely.

Reply:
Alternatively, there are a bunch of utilities to narrow these types: https://github.com/ChristianMurphy/unifiedjs.github.io/blob/docs/typescript-guide-and-recipes/doc/learn/node-type-narrowing-in-typescript.md.
You can do this type safe!

Reply:
> But now that I think about it, if `isValidLink` is true, then `isText` will always be false so we could probably remove the `isText` completely.

@kentcdodds We should keep the `isText` check as we're also transforming text that's not a valid link apparently 🤔